diff -Nur srp-2.1.1.orig/base/pam_eps/Makefile.in srp-2.1.1.cflags/base/pam_eps/Makefile.in
--- srp-2.1.1.orig/base/pam_eps/Makefile.in	2006-08-05 03:35:32.000000000 +0200
+++ srp-2.1.1.cflags/base/pam_eps/Makefile.in	2006-08-05 03:36:11.000000000 +0200
@@ -28,6 +28,7 @@
 
 CPPFLAGS = @CPPFLAGS@ -I../../libsrp $(USE_LCKPWDF) $(NEED_LCKPWDF) $(EXTRAS)
 CFLAGS = @CFLAGS@
+LDFLAGS = @LDFLAGS@
 
 DISTFILES = Makefile.in Makefile.unix README README.pam-eps README.rpm \
   SOURCES SPECS pam.conf.eps pam.conf.migrate \
@@ -81,13 +82,13 @@
 $(LIBOBJD): $(LIBSRC)
 
 $(LIBAUTHSH):	$(LIBAUTHSRC) $(LIBOBJD)
-		$(CC) $(CCSHARED) -o $@ $(LIBAUTHOBJD) $(LIBS)
+		$(CC) $(LDFLAGS) $(CCSHARED) -o $@ $(LIBAUTHOBJD) $(LIBS)
 
 $(LIBPASSWDSH):	$(LIBPASSWDSRC) $(LIBOBJD)
-		$(CC) $(CCSHARED) -o $@ $(LIBPASSWDOBJD) $(LIBS)
+		$(CC) $(LDFLAGS) $(CCSHARED) -o $@ $(LIBPASSWDOBJD) $(LIBS)
 
 $(CHKPWD):	eps_chkpwd.c
-		$(CC) -o $@ eps_chkpwd.c $(STATIC) $(LIBS)
+		$(CC) $(LDFLAGS) $(CFLAGS) -o $@ eps_chkpwd.c $(STATIC) $(LIBS)
 
 install: all
 	mkdir -p $(FAKEROOT)$(SECUREDIR)
